geopandas/geodataframe.py

```diff
@@ -190,6 +190,10 @@ def __init__(self, data=None, *args, geometry=None, crs=None, **kwargs):
                 and not geometry.crs == crs
             ):
                 raise ValueError(crs_mismatch_error)
+            # If geometry kwarg is an array, set the name of the active
+            # geometry column to geometry
+            if not isinstance(geometry, str):
+                self._geometry_column_name = "geometry"
```

Comment:
Is this needed here if `set_geometry` also has logic to use "geometry" in case of DEFAULT_GEO_COLUMN_NAME